src/Aspire.Hosting/Dcp/DcpExecutor.cs

```diff
@@ -1239,14 +1247,18 @@ async Task CreateContainerAsyncCore(AppResource cr, CancellationToken cancellati
             // Create a custom container network for Aspire if there are container resources
             if (containerResources.Any())
             {
-                var network = ContainerNetwork.Create(DefaultAspireNetworkName);
+                var network = ContainerNetwork.Create(DefaultAspireNetworkResource);
                 if (containerResources.Any(cr => cr.ModelResource.GetContainerLifetimeType() == ContainerLifetime.Persistent))
                 {
                     // If we have any persistent container resources
                     network.Spec.Persistent = true;
                     // Persistent networks require a predictable name to be reused between runs.
                     // Append the same project hash suffix used for persistent container names.
-                    network.Spec.NetworkName = $"{DefaultAspireNetworkName}-{_nameGenerator.GetProjectHashSuffix()}";
+                    network.Spec.NetworkName = $"{DefaultAspirePersistentNetworkName}-{_nameGenerator.GetProjectHashSuffix()}";
```

Comment:
if we're doing name updates here, do we think it's worth adding the apphost application name in here as well to make it easier to identify which apphost a network belongs to?

Reply:
It's worth considering, but we'd have to do some normalization to account for invalid characters (and possibly maximum length).

Reply:
So running TestShop.AppHost, you'll get a network named `aspire-persistent-network-<hash>-TestShop`.